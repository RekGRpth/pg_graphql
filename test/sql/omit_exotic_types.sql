begin;

    /*
        Composite and array types are not currently supported as inputs
        - confirm composites are not allowed anywhere
        - confirm arrays are not allowed as input
    */

    create type complex as (r int, i int);

    create table something(
        id serial primary key,
        name varchar(255) not null,
        tags text[],
        comps complex,
        js json,
        jsb jsonb
    );

    select graphql.rebuild_schema();

    select
        name, parent_type, type_, is_not_null, is_array, is_arg
    from
        graphql.field
    where
        entity = 'something'::regclass
        and column_name is not null
    order by
        parent_type,
        name;

rollback;
