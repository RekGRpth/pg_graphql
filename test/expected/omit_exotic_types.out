begin;
    /*
        Composite and array types are not currently supported as inputs
        - confirm composites are not allowed anywhere
        - confirm arrays are not allowed as input
    */
    create type complex as (r int, i int);
    create table something(
        id serial primary key,
        name varchar(255) not null,
        tags text[],
        comps complex,
        js json,
        jsb jsonb
    );
    select graphql.rebuild_schema();
 rebuild_schema 
----------------
 
(1 row)

    select
        name, parent_type, type_, is_not_null, is_array, is_arg
    from
        graphql.field
    where
        entity = 'something'::regclass
        and column_name is not null
    order by
        parent_type,
        name;
 name |     parent_type      |      type_       | is_not_null | is_array | is_arg 
------+----------------------+------------------+-------------+----------+--------
 id   | Something            | Int              | t           | f        | f
 js   | Something            | JSON             | f           | f        | f
 jsb  | Something            | JSON             | f           | f        | f
 name | Something            | String           | t           | f        | f
 tags | Something            | String           | f           | t        | f
 id   | SomethingFilter      | IntFilter        | f           | f        | f
 name | SomethingFilter      | StringFilter     | f           | f        | f
 js   | SomethingInsertInput | JSON             | f           | f        | t
 jsb  | SomethingInsertInput | JSON             | f           | f        | t
 name | SomethingInsertInput | String           | f           | f        | t
 id   | SomethingOrderBy     | OrderByDirection | f           | f        | f
 name | SomethingOrderBy     | OrderByDirection | f           | f        | f
 tags | SomethingOrderBy     | OrderByDirection | f           | f        | f
 js   | SomethingUpdateInput | JSON             | f           | f        | t
 jsb  | SomethingUpdateInput | JSON             | f           | f        | t
 name | SomethingUpdateInput | String           | f           | f        | t
(16 rows)

rollback;
